### ISSUE TITLE
java class output doesn't show in cider-repl buffer

### ISSUE THREAD

**Issue by sound2gd** [clojure-emacs/cider#1896]
## Expected behavior
print the `something` and `nil`
## Actual behavior
print the `nil`,`something` is gone
## Steps to reproduce the problem

```clojure
(.println (java.lang.System/out) "something")
```

### CIDER version information
```
;; CIDER 0.15.0snapshot (package: 20161201.757), nREPL 0.2.12
;; Clojure 1.8.0, Java 1.8.0_101
```

### Lein/Boot version

Leiningen 2.7.1 on Java 1.8.0_101 Java HotSpot(TM) 64-Bit Server VM 

### Emacs version

25.1.1

### Operating system

Mac OS X Sierra

**Comment by bhagany** [comment]
This is pretty much the same issue as #1588.  I have a proof-of-concept fix here: https://github.com/bhagany/cider-nrepl/tree/repl-err.  Hopefully soon to be a PR.

**Comment by bhagany** [comment]
Also, I'm currently stuck on the fact that the printed output gets printed before the prompt that causes it - that is, the prompt moves down and the output prints above it. If anyone has a solution to that, please let me know.

**Comment by bhagany** [comment]
This is fixed with #387
